include/boost/gil/extension/dynamic_image/apply_operation_base.hpp

```diff
@@ -163,6 +165,108 @@ static typename Op::result_type BOOST_FORCEINLINE apply_operation_base(const Bit
+            }                                                                                           \
+            throw;                                                                                      \
+        }                                                                                               \
+        template <typename Types, typename Info, typename Bits, typename UnaryOp>                                     \
+        auto apply(Bits& bits, std::size_t index, const Info& info, UnaryOp op) const {        \
+            typedef typename mpl::begin<Types>::type                                             \
+            BOOST_PP_REPEAT(N, GIL_FWD_TYPEDEFS, BOOST_PP_EMPTY)                                            \
+            T##N;                                                                                       \
+            switch (index) {                                                                            \
+                BOOST_PP_REPEAT(BOOST_PP_ADD(N,1), GIL_FWD_CASE_WITH_INFO, BOOST_PP_EMPTY)                            \
+            }                                                                                           \
+            throw;                                                                                      \
+        }                                                                                               \
+        template <typename Types, typename Bits, typename Info, typename UnaryOp>                                     \
+        auto applyc(const Bits& bits, std::size_t index, const Info& info, UnaryOp op) const { \
+            typedef typename mpl::begin<Types>::type                                             \
+            BOOST_PP_REPEAT(N, GIL_FWD_TYPEDEFS, BOOST_PP_EMPTY)                                            \
+            T##N;                                                                                       \
+            switch (index) {                                                                            \
+                BOOST_PP_REPEAT(BOOST_PP_ADD(N,1), GIL_FWD_CONST_CASE_WITH_INFO,BOOST_PP_EMPTY)                       \
+            }                                                                                           \
+            throw;                                                                                      \
+        }                                                                                               \
+    };
+
+#define GIL_GENERATE_APPLY_FWD_OPS(N) BOOST_PP_REPEAT(N, GIL_APPLY_FWD_OP, BOOST_PP_EMPTY)
+
+namespace detail {
+	template <std::size_t N> struct apply_operation_fwd_fn {};
+
+	// Create specializations of apply_operation_fn for each N 0..100
+	GIL_GENERATE_APPLY_FWD_OPS(99)
+} // namespace detail
+
+// unary application
+template <typename Types, typename Bits, typename Op>
+auto BOOST_FORCEINLINE apply_operation_basec(const Bits& bits, std::size_t index, Op op) {
```

Comment:
Could we try to keep the lines shorter, this way?
```
auto
BOOST_FORCEINLINE
apply_operation_basec(const Bits& bits, std::size_t index, Op op)
{
...
}
```

BTW, there have been some discussion about possibility to get rid of `BOOST_FORCEINLINE` one day.

Reply:
@sdebionne Sorry, I meant this, of course

```
BOOST_FORCEINLINE
auto apply_operation_basec(const Bits& bits, std::size_t index, Op op)
```